code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -304,6 +331,88 @@
 		busy = 0
 		stop_automated_movement = 0
 
+/mob/living/simple_animal/hostile/giant_spider/verb/web()
+	set name = "Spin Web"
+	set desc = "Create a web that slows down movement."
+	set category = "Greimorian"
+
+	var/obj/effect/spider/stickyweb/W = locate() in get_turf(src)
+	if(!W)
+		src.visible_message("\The [src] begins to secrete a sticky substance.")
+		if(!do_after(src, 10)) return
+		new /obj/effect/spider/stickyweb(src.loc)
+
+
+/mob/living/simple_animal/hostile/giant_spider/nurse/verb/cocoon()
+	set name = "Cocoon and Feed"
+	set desc = "Cocoon an incapacitated mob so you can feed upon it. This will give you one food point."
+	set category = "Greimorian"
+
+	for(var/mob/living/P in range(1,src))
```

Comment:
1) check if it is a robotic mob
2) make it a selection of a mob instead of picking a random one (so, by building a list and then using tgui_input_list() to select a mob)

Reply:
Handled